sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -1526,106 +1526,136 @@ object QueryExecutionErrors {
   }
 
   def unsupportedOperationExceptionError(): Throwable = {
-    new UnsupportedOperationException
+    new SparkUnsupportedOperationException(
+      errorClass = "UNSUPPORTED_OPERATION",
```

Comment:
The lack of error message here is really bad for the user, although this is technically an internal error (we should never call `defaultConcreteType` on `AnyDataType`). When  https://github.com/apache/spark/pull/34123 merges, can you use `INTERNAL_ERROR` instead, and provide a meaningful error message? In this case, we should also rename the class.